lib/binary_tree.c

```diff
@@ -8,6 +12,9 @@ struct bt_node *bt_new_node(int val)
     if (bt_node_ptr) {
         bt_node_ptr->data = val;
         bt_node_ptr->left = bt_node_ptr->right = NULL;
+    } else {
```

Comment:
(style) Better to return an error in "if" clause than "else".

struct bt_node *bt_node_ptr = malloc(sizeof(struct bt_node));
      if (bt_node_ptr == NULL) {
          fprintf("ERROR");
          return error; //or goto out to keep single exit point
      }
      //contines here